_studio/enctools/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+set(MFX_ORIG_LDFLAGS "${MFX_LDFLAGS}" )
+
+mfx_include_dirs( )
+
+include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/include )
+include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/aenc/include )
+include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/shared/include)
+
+set(sources
+    ${CMAKE_CURRENT_SOURCE_DIR}/enctools.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/mfx_enctools_brc.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/mfx_enctools_common.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/mfx_enctools_aenc.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/mfx_enctools_utils.cpp
+    )
+set( sources.plus "" )
+
+set( defs "${API_FLAGS} ${WARNING_FLAGS}" )
+
+if (MFX_ENABLE_AENC)
+  find_library( AENC_LIBRARY libaenc.a PATHS ${CMAKE_HOME_DIRECTORY}/contrib/aenc/ )
+  if(NOT ${AENC_LIBRARY} MATCHES NOTFOUND)
+    message(STATUS "aenc library was found here ${AENC_LIBRARY}")
+
+    add_library(aenc STATIC IMPORTED GLOBAL)
+    set_target_properties(aenc PROPERTIES IMPORTED_LOCATION "${CMAKE_HOME_DIRECTORY}/contrib/aenc/libaenc.a")
+    target_include_directories(aenc INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/include)
+  else()
+    message(STATUS "aenc library was not found (optional)! mfxlib will be built without aenc")
+    set(MFX_ENABLE_AENC OFF)
+  endif()
+endif()
+
+if (MFX_ENABLE_AENC)
```

Comment:
you don't need this block at all. Actually, add `#cmakedefine MFX_ENABLE_AENC` in the mfxconfig.h.in and what you are doing will be done automatically and in a more clear way.

Reply:
What if MFX_ENABLE_AENC=ON is set at build but libaenc.a is not found? We need to have  MFX_ENABLE_AENC undefined  in the code in this case. In other words, we need to have a define defined/undefined depending on the presence of libanenc.a

Reply:
Or we just don't need to consider such a situation?

Reply:
We just need to stop with FATAL_ERROR message on cmake configuration stage.